plugins/cost-insights/src/components/CostOverviewCard/CostOverviewByProduct.tsx

```diff
@@ -0,0 +1,211 @@
+  dailyCostData,
+}: CostOverviewByProductProps) => {
+  const theme = useTheme<CostInsightsTheme>();
+  const styles = useStyles(theme);
+  const lastCompleteBillingDate = useLastCompleteBillingDate();
+  const { duration } = useFilters(mapFiltersToProps);
+  const groupedCosts = dailyCostData.groupedCosts;
+
+  if (!groupedCosts) {
+    return null;
+  }
+
+  const flattenedAggregation = groupedCosts
+    .map(cost => cost.aggregation)
+    .flat();
+  const totalCost = flattenedAggregation.reduce(
+    (acc, agg) => acc + agg.amount,
+    0,
+  );
+
+  const productsByDate = groupedCosts.reduce((prodByDate, group) => {
+    const product = group.id;
+    group.aggregation.forEach(curAggregation => {
+      const productCostsForDate = prodByDate[curAggregation.date] || {};
+      // Group products with less than 10% of the total cost into "Other" category
+      if (aggregationSum(group.aggregation) < totalCost * LOW_COST_THRESHOLD) {
+        prodByDate[curAggregation.date] = {
+          ...productCostsForDate,
+          Other:
+            (prodByDate[curAggregation.date].Other || 0) +
+            curAggregation.amount,
+        };
+      } else {
+        prodByDate[curAggregation.date] = {
+          ...productCostsForDate,
+          [product]: curAggregation.amount,
+        };
+      }
+    });
+
+    return prodByDate;
+  }, {} as Record<string, Record<string, number>>);
+
+  const chartData: any[] = Object.keys(productsByDate).map(date => {
+    return {
+      ...productsByDate[date],
+      date: Date.parse(date),
+    };
+  });
+
+  const tooltipRenderer: ContentRenderer<TooltipProps> = ({
+    label,
+    payload = [],
+  }) => {
+    if (isInvalid({ label, payload })) return null;
+
+    const title = dayjs(label).utc().format(DEFAULT_DATE_FORMAT);
+    const items = payload.map(p => ({
+      label: p.dataKey as string,
+      value: formatGraphValue(p.value as number),
+      fill: p.fill!,
+    }));
+
+    return (
+      <Tooltip title={title}>
+        {items.reverse().map((item, index) => (
+          <TooltipItem key={`${item.label}-${index}`} item={item} />
+        ))}
+      </Tooltip>
+    );
+  };
+
+  const previousPeriodTotal = getPreviousPeriodTotalCost(
+    flattenedAggregation,
+    duration,
+    lastCompleteBillingDate,
+  );
+  const currentPeriodTotal = totalCost - previousPeriodTotal;
+
+  const renderAreas = () => {
+    const highCostGroups = groupedCosts.filter(
+      group =>
+        aggregationSum(group.aggregation) >= totalCost * LOW_COST_THRESHOLD,
+    );
+    const sortedCostGroups = highCostGroups
+      .sort(
+        (a, b) => aggregationSum(a.aggregation) - aggregationSum(b.aggregation),
+      )
+      .map(group => group.id);
+    // Keep 'Other' category at the bottom of the stack
+    return ['Other', ...sortedCostGroups].map((group, i) => (
+      <Area
+        dataKey={group}
+        stackId="1"
+        fillOpacity="1"
+        stroke={DataVizColors[i]}
```

Comment:
@timbonicus is this to keep the highest cost product the same color ? Yeah, I figured that it wasn't possible to need more than 10 colors since we can't have more than 10 products with >= 10% of costs

Reply:
Yeah, for the DataViz colors they should go in declared order - so the highest cost should always be blue.